pkg/scheduler/schedule_one.go

```diff
@@ -437,6 +437,11 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 		if !s.IsUnschedulable() {
 			return nil, diagnosis, s.AsError()
 		}
+		// All nodes will have the same status. Some non trivial refactoring is
+		// needed to avoid this copy.
+		for _, n := range allNodes {
+			diagnosis.NodeToStatusMap[n.Node().Name] = s
+		}
 		// Record the messages from PreFilter in Diagnosis.PreFilterMsg.
 		msg := s.Message()
 		diagnosis.PreFilterMsg = msg
```

Comment:
Could this cause the message to be duplicated?

Reply:
Good question. @sanposhiho could you double check the event/log is still rendered properly?

Reply:
It's a good catch. I changed type.go so that we won't make duplicates in the message.